artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/management/AddressControl.java

```diff
@@ -150,6 +150,26 @@ String sendMessage(@Parameter(name = "headers", desc = "The headers to add to th
                       @Parameter(name = "user", desc = "The user to authenticate with") String user,
                       @Parameter(name = "password", desc = "The users password to authenticate with") String password) throws Exception;
 
+   /**
+    * @param headers  the message headers and properties to set. Can only
+    *                 container Strings maped to primitive types.
+    * @param body     the text to send
+    * @param durable
+    * @param user
+    * @param password @return
+    * @param createUserId whether or not to auto generate a userID, used as a Message ID in JMS
+    * @throws Exception
+    */
+   @Operation(desc = "Sends a TextMessage to a password-protected address.", impact = MBeanOperationInfo.ACTION)
+   String sendMessage(@Parameter(name = "headers", desc = "The headers to add to the message") Map<String, String> headers,
+                      @Parameter(name = "type", desc = "A type for the message") int type,
+                      @Parameter(name = "body", desc = "The body (byte[]) of the message encoded as a string using Base64") String body,
+                      @Parameter(name = "durable", desc = "Whether the message is durable") boolean durable,
+                      @Parameter(name = "user", desc = "The user to authenticate with") String user,
+                      @Parameter(name = "password", desc = "The users password to authenticate with") String password,
+                      @Parameter(name = "createUserId", desc = "whether or not to auto generate a userID, used as a Message ID in JMS") boolean createUserId) throws Exception;
```

Comment:
Why not just say 'addMessageId'? Contiuning to reference this 'userId' concept, when its absolutely about a message-id, and definitely not a userId, and then having to document that it is a message id, just seems unnecessarily confusing.